### ISSUE TITLE
Problems with road bike/fastbike profile

### ISSUE THREAD

**Issue by tbsmark86** [abrensch/brouter#479]
Problem a)
Routing in Norway along the E6. This is marked as highway=trunk. Problem is in norther parts roads are scares detours are long and may require ferries and traffic on this 'trunk' road is actually low.
This case has a detour of about 320km while the trekking profile simply uses the 'bad' trunk road. In this specific case the road is part of EuroVelo 11.
[Road Bike: 406km](https://brouter.de/brouter-web/#map=8/69.875/25.606/osm-mapnik-german_style,route-quality&lonlats=24.923397,70.073119;25.538175,69.458999&profile=fastbike)
[Trekking Bike: 82km](https://brouter.de/brouter-web/#map=8/69.865/25.651/osm-mapnik-german_style,route-quality&lonlats=24.923397,70.073119;25.538175,69.458999)

Cost is for example:
   cost: 10000	tags: highway=trunk surface=asphalt route_bicycle_icn=yes

Idea:
```
-  switch or highway=trunk highway=trunk_link          switch allow_motorways 1.5 10
+  switch or highway=trunk highway=trunk_link          switch or allow_motorways any_cycleroute 1.5 4
```
Problem b)
Avoiding of highway=path independent cycleways.
Here an example of a very iconic cyclepath in Berlin (so called "Krone"). 
Given that result its hard to argue with anyone why he should use brouter instead of say komoot. It's just outright stupid.
[Road Bike: 15km passing through city](https://brouter.de/brouter-web/#map=13/52.4596/13.2221/osm-mapnik-german_style,route-quality&lonlats=13.295051,52.506818;13.175187,52.420115&profile=fastbike)
[Trekking Bike: 13.5km using cyclepath](https://brouter.de/brouter-web/#map=13/52.4684/13.2484/osm-mapnik-german_style,route-quality&lonlats=13.295051,52.506818;13.175187,52.420115)

Problem is that it's marked as highway=path:
  cost: 2000	tags:  highway=path surface=asphalt foot=designated bicycle=designated smoothness=excellent
Looking at the OSM wiki for highway value that seems correct because it's (almost) exclusively for cycling/walking. But in this "path" is really wide (between 1 and 2 car lanes).

My Idea:
```
+assign isgoodpaved or surface=asphalt or surface=concrete surface=paving_stones
+assign isverygoodpaved and isgoodpaved or smoothness=excellent smoothness=good

-   switch or bicycle=designated bicycle_road=yes 2.0
-   switch ispaved 2.0 100.0
+   switch or bicycle=designated bicycle_road=yes 
+     switch isverygoodpaved 1.1
+     switch isgoodpaved 1.3
+     switch ispaved 1.7
+     2.0
+   switch isgoodpaved 1.7 switch ispaved 2.0 100.0
```
And similar rules for highway=service and other.

Here are two more similar but less iconic example. Really fast road bikes users (35kph+) might actually prefer the current result:
[Road Bike: 9.4km](https://brouter.de/brouter-web/#map=13/52.1963/12.3129/osm-mapnik-german_style,route-quality&lonlats=12.374983,52.173879;12.291384,52.213234&profile=fastbike)
[Trekking Bike: 8.9km](https://brouter.de/brouter-web/#map=14/52.2000/12.3557/osm-mapnik-german_style,route-quality&lonlats=12.374983,52.173879;12.291384,52.213234)
This is a new build cycle path on an old railway track.

[Road Bike: 14.5km](https://brouter.de/brouter-web/#map=12/52.9846/13.3386/osm-mapnik-german_style,route-quality&lonlats=13.393407,52.869622;13.339725,52.976762&profile=fastbike)
[Trekking Bike: 14.3km](https://brouter.de/brouter-web/#map=12/52.9846/13.3386/osm-mapnik-german_style,route-quality&lonlats=13.393407,52.869622;13.339725,52.976762)
In this case its a service way along the canal part, part of EuroVelo 7 and others.

P.s. why are there 5 options in fastbike.brf that are not used at all? (allow_steps, allow_ferries, ignore_cycleroutes, stick_to_cycleroutes, avoid_unsafe)

**Comment by EssBee59** [comment]
hello tbsmark,

to your "problem b":
If you ask 10 fastbikers which route they prefer, you will at least get 5 different responses ....
(I had also long discussions with my friends about using or not pathes)

But using Brouter you are able to create the exact profile you prefer!
I did so and created my own profile "fastbile very low traffic". On my test instance you can have a look at the calculated route in your examples.
http://brouter.de/essbee/#map=12/52.4657/13.2313/osm-mapnik-german_style,route-quality&lonlats=13.295051,52.506818;13.175187,52.420115

(you can further adapt the option "consider_traffic" in the profile)

hope it helps

**Comment by tbsmark86** [comment]
Yes I know you can create your own profile. But it's rather impractical unless you also have your own server running.

It would be nice to have a sane default.

I'am adding  Problem c)
The costs of the road bike profile tend to use the road instead of the cyclingpath along the road - even if it's mandatory by law to use the cyclepath.
In theory the road should be tagged bicycle=use_sidepath. Not sure how correct the OSM data is starting with the fact that different people might use the law different.
The "Krone" example from b) contains such bit on the current fastbike profile:
[Motorway Junction](https://brouter.de/brouter-web/#map=14/52.4305/13.2130/osm-mapnik-german_style,route-quality&lonlats=13.206079,52.424774;13.184602,52.418957&profile=fastbike)  you have to use the tunnel here.

Interesting: If i use bicycle=use_sidepath in bikeaccess as a 'no' the fastbike profile creates an even worse route with 1km of bad cobbles. Instead of using the tunnel which is ok aside from the extra climbing.
Reason is - again - highway=path with cost of 2000 for the tunnel bit.

It's mostly annoying because your gps device might start beeping that your of the course. But it could also change the route if the cyclepath is particularly bad.

**Comment by EssBee59** [comment]
> Yes I know you can create your own profile. But it's rather impractical unless you also have your own server running.

No, you do not nead your own server to create a custom profile. 
(A current restriction is, you can not easily store/resuse the created profile, but this restriction could be eliminated soon, see https://github.com/nrenner/brouter-web/issues/472 )

As explained, a standard / generic profile is not possible for ALL fastbikers...
Did you test my profile ?
 http://brouter.de/essbee/#map=12/52.4657/13.2313/osm-mapnik-german_style,route-quality&lonlats=13.295051,52.506818;13.175187,52.420115 
 
You can also test the profile developped by Marcus (it is installed / available on his server):
https://brouter.m11n.de/#map=12/52.4657/13.2313/osm-mapnik-german_style,route-quality&lonlats=13.295051,52.506818;13.175187,52.420115

**Comment by polyscias** [comment]
> In theory the road should be tagged bicycle=use_sidepath.

If it's indeed mandatory by law to use the cyclepath why do you not edit the OpenStreetMap data yourself?

Things like adding "bicycle=use_sidepath" is pretty simple, open https://www.openstreetmap.org and zoom to the area. Then press the Edit button (left-top) and select the iD editor. Go to Tags and add bicycle=use_sidepath. That will not only benefit you but also others.

Looking at [your example](https://brouter.de/brouter-web/#map=14/52.4305/13.2130/osm-mapnik-german_style,route-quality&lonlats=13.206079,52.424774;13.184602,52.418957&profile=fastbike), it does not help you indicate you like to start and on the main road and you are using the fastbike profile instead of the Trekking profile.

**Comment by tbsmark86** [comment]
> why do you not edit the OpenStreetMap data yourself?

Because it's already present in the above example. But neither the fastbike nor the trekking profile do check for it.

> Looking at your example, it does not help you indicate you like to start and on the main

Why? Assuming the profile try to create legal routes (does it?) my starting point is perfectly legal. The mandatory cycle path starts 50m later.


> As explained, a standard / generic profile is not possible for ALL fastbikers...

Yes. But Is that an excuse to fix a case the 95% will dislike?
A solution might be to use the 'stick_to_cycleroutes' option seeing as it's currently a no-op.


Ok Questions: Are profiles part of this project or are they more like examples and everybody is expected anyway to create their own? In later case my issue is moot.

**Comment by polyscias** [comment]
>> why do you not edit the OpenStreetMap data yourself?
>
> Because it's already present in the above example. But neither the fastbike nor the trekking profile do check for it.

You are 90% correct that bicycle=use_sidepath is already present for the route I just found out but it is missing on these two pieces of road: [385126058](https://www.openstreetmap.org/way/385126058) and [912706185](https://www.openstreetmap.org/way/912706185)

That the fastbike profile does not check for it is maybe an oversight, I do not know if the German DE:241 traffic sign indicates also mandatory for fastbikes. A fastbike is in OSM term's is also no bicycle but more likely a moped.

For the trekking profile, bicycle=use_sidepath is taken into account, see #79 and more in particular [here](https://github.com/abrensch/brouter/blob/eba0b18689d03d0ff3611a8896c4b593e45d8d15/misc/profiles2/trekking.brf#L122). If I take exactly your route and change the profile to Trekking [the route takes the cycleway](https://brouter.de/brouter-web/#map=14/52.4186/13.1990/standard,route-quality&lonlats=13.206079,52.424774;13.184602,52.418957).

> Are profiles part of this project or are they more like examples and everybody is expected anyway to create their own? In later case my issue is moot.

There are some "reference" profiles part of the project but there are quite some other published profiles around:

- [Poutnikl's profiles](https://github.com/poutnikl/Brouter-profiles)
- [Zossebart's profiles](https://github.com/zossebart/brouter-mtb)
- [Thomas Traber's velomobile profile](https://github.com/ThomasTraber/brouter_profiles_and_testing)
- [Jacob's city/street profile](https://github.com/utack/utack_brouter_data)
- [FFMbyBicycle's profile](https://github.com/FFMbyBicycle/brouter-cycling-profiles)
- [AFischer's waterway profiles](https://github.com/afischerdev/brouter-profile)
- [BRouter profile für den CXBerlin Routenplaner](https://github.com/cxberlin/brouter-profiles) 
- [utack's profile](https://github.com/utack/utack_brouter_profile)
- The EssBee Fastbike (very low traffic) profile
- Marcus's Roadbike (minimized traffic) profile
- Profiles from http://www.velomobilforum.de
-  The Fastbike Asia profile.

This last profile has already what you asked for under "Problem a)"

There will be much more unpublished profiles around.

**Comment by EssBee59** [comment]
> Ok Questions: Are profiles part of this project or are they more like examples and everybody is expected anyway to create their own? In later case my issue is moot.

Hello Tobias,
Again, it is difficult to provide a generic fastbike profile, valid for all countries,  for each fastbiker and ... any situation:
As example in Germany a biker have to use an existing cycleway, but if you are in a group of more than 15 bikers, the group can use the road (2 bikers parallel is also allowed)
Hope, you see with this example that you haveto choose or create the right profile for you.

**Comment by tbsmark86** [comment]
Ok.  I see.

>  A fastbike is in OSM term's is also no bicycle but more likely a moped.

Really? Well than the label/translation in brouter-web is wrong it says "Road Bike" .

In that case its maybe more of an S-Pedelec (E-Bike upto 45 kph) which has the opposite problem: There are forbidden on cycling paths in Germany.

> For the trekking profile, bicycle=use_sidepath is taken into accoun

Sorry missed that.

**Comment by zod** [comment]
I'm reopening this because I think your suggestions are valuable and I'd like to clarify a few things from my point of view.

> P.s. why are there 5 options in fastbike.brf that are not used at all? (allow_steps, allow_ferries, ignore_cycleroutes, stick_to_cycleroutes, avoid_unsafe)

`fastbike.brf` is derived from some other profile and hasn't been cleaned up.

> Ok Questions: Are profiles part of this project or are they more like examples and everybody is expected anyway to create their own? In later case my issue is moot.

We try to provide profiles that work out of the box for most users. Quite often tweaking parameters causes regressions for other users. If it's possible to improve the profile for some region without unwanted side effects we'll do it.

> A fastbike is in OSM term's is also no bicycle but more likely a moped.

The [comment at the beginning of fastbike.brf](https://github.com/abrensch/brouter/blob/master/misc/profiles2/fastbike.brf) is a bit ambiguous as it mentions both roadbike and s-pedelec for which at least in Germany the rules are totally different. The term *fastbike* is only used by BRouter, it's no English term or used within OSM.

I think *fastbike* should target roadbike or velomobiles/recumbent bikes. For S-pedelecs it's hard to create a generic profile because the rules are different in each country.

**Comment by EssBee59** [comment]
> I think _fastbike_ should target roadbike or velomobiles/recumbent bikes. For S-pedelecs it's hard to create a generic profile because the rules are different in each country.

Hello,

I thought "fastbike" ist in german "Rennrad", in french "velo de course".
(a very fast bike with special /short tires which are only usable on asphalt!)

**Comment by zod** [comment]
Not according to some online dictionaries ([dict.leo.org](https://dict.leo.org/german-english/rennrad), [dict.cc](https://www.dict.cc/?s=Rennrad)) or [Wikipedia](https://en.wikipedia.org/wiki/Racing_bicycle). It's a term that's specific to BRouter which tries to specify a common profile for several types of fast bikes which share common restrictions.

**Comment by tbsmark86** [comment]
>  Quite often tweaking parameters causes regressions

Well yes that's the problem with highway=path I fear. Therefore my idea with surface=asphalt and smoothness.
Problem with highway=path is that it can be very different in quality and width.

To help understand my complaint b) better here is a  [Random YT Video](https://www.youtube.com/watch?v=LJO69tQuh2Y) of this highway=path.
Of course its really rare to have a highway=path of this quality.
If anybody complains that highway=path is not appropriate ... Well because this is cycling and walking only then according to osm-de wiki that is in correct (as far as I understand the wiki).

Obviously there are also lots of highway=path that are definitely not fastbike and maybe not even trekkingbike even if bicycle=yes - because those might be MTB single trails.

**Comment by tbsmark86** [comment]
> fastbike.brf is derived from some other profile and hasn't been cleaned up.

Ok, at least two of them i would suggest fixing instead of removing - and without any regression i hope:

```
-  switch    highway=steps                             1000
-  switch    route=ferry                               5.67
+  switch    highway=steps                             switch allow_steps 40 1000
+  switch    route=ferry                               switch allow_ferries 5.67 10000
```
